File: services/user-service/src/main/java/org/example/services/user/service/AuthService.java
package org.example.services.user.service;

import org.example.auth.*;

public interface AuthService {

    void register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    String verifyUser(String token);

    void forgotPassword(ForgotPasswordRequest request);

    void resetPassword(ResetPasswordRequest request);

    AuthResponse refreshToken(String email);
}

File: frontend/admin-portal/src/services/authService.js
import axiosClient from '../config/axiosConfig';

const authService = {
    login: (credentials) => {
        // credentials: { email, password }
        return axiosClient.post('/auth/login', credentials);
    },

    register: (data) => {
        // data: { fullName, email, password }
        return axiosClient.post('/auth/register', data);
    },

    verifyEmail: (token) => {
        return axiosClient.get(`/auth/verify?token=${token}`);
    },

    forgotPassword: (email) => {
        return axiosClient.post('/auth/forgot-password', { email });
    },

    resetPassword: (token, newPassword) => {
        return axiosClient.post('/auth/reset-password', { token, newPassword });
    },
};

export default authService;